### PR TITLE
fix(onboarding/pr): include retry/refresh checkbox message only if onboardingRebaseCheckbox is true

### PR DESCRIPTION
**Pull request by aslafy-z**
## Changes

Updated the default onboarding PR description template to provide clearer instructions. Now explicitly mentions that the retry/rebase checkbox must be ticked for the custom configuration to be considered on next Renovate run and for it to update the PR description.
This applies when the `onboardingRebaseCheckbox` config option is enabled.

## Context

It is not obvious that the box should be ticked.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by rarkins** [comment]
I thought that rebase/retry will erase any custom config. @RahulGautamSingh

**Comment by aslafy-z** [comment]
It looks like it does not 🤷

File: lib/workers/repository/onboarding/pr/__snapshots__/config-description.spec.ts.snap
Comment (HonkingGoose) [inline lib/workers/repository/onboarding/pr/__snapshots__/config-description.spec.ts.snap:12]:
```suggestion
🔡 Would you like to change the way Renovate is upgrading your dependencies? Simply edit the \`.github/renovate.json\` in this branch with your custom config and select the Retry/Rebase checkbox below, the list of Pull Requests in the "What to Expect" section below will be updated the next time Renovate runs.
```

File: lib/workers/repository/onboarding/pr/__snapshots__/config-description.spec.ts.snap
Comment (HonkingGoose) [inline lib/workers/repository/onboarding/pr/__snapshots__/config-description.spec.ts.snap:27]:
```suggestion
🔡 Would you like to change the way Renovate is upgrading your dependencies? Simply edit the \`renovate.json\` in this branch with your custom config and select the Retry/Rebase checkbox below, the list of Pull Requests in the "What to Expect" section below will be updated the next time Renovate runs.
```

File: lib/workers/repository/onboarding/pr/__snapshots__/config-description.spec.ts.snap
Comment (HonkingGoose) [inline lib/workers/repository/onboarding/pr/__snapshots__/config-description.spec.ts.snap:42]:
```suggestion
🔡 Would you like to change the way Renovate is upgrading your dependencies? Simply edit the \`renovate.json\` in this branch with your custom config and select the Retry/Rebase checkbox below, the list of Pull Requests in the "What to Expect" section below will be updated the next time Renovate runs.
```

File: lib/workers/repository/onboarding/pr/__snapshots__/config-description.spec.ts.snap
Comment (HonkingGoose) [inline lib/workers/repository/onboarding/pr/__snapshots__/config-description.spec.ts.snap:60]:
```suggestion
🔡 Would you like to change the way Renovate is upgrading your dependencies? Simply edit the \`renovate.json\` in this branch with your custom config and select the Retry/Rebase checkbox below, the list of Pull Requests in the "What to Expect" section below will be updated the next time Renovate runs.
```

File: lib/workers/repository/onboarding/pr/config-description.spec.ts
Comment (HonkingGoose) [inline lib/workers/repository/onboarding/pr/config-description.spec.ts:51]:
```suggestion
        🔡 Would you like to change the way Renovate is upgrading your dependencies? Simply edit the \`renovate.json\` in this branch with your custom config and select the Retry/Rebase checkbox below, the list of Pull Requests in the "What to Expect" section below will be updated the next time Renovate runs.
```

File: lib/workers/repository/onboarding/pr/config-description.ts
Comment (HonkingGoose) [inline lib/workers/repository/onboarding/pr/config-description.ts:57]:
```suggestion
  desc += ` Simply edit the \`${configFile}\` in this branch with your custom config and select the Retry/Rebase checkbox below, the list of Pull Requests in the "What to Expect" section below will be updated the next time Renovate runs.`;
```

**Review by HonkingGoose — CHANGES_REQUESTED** [review]
I prefer the verb `select` over `tick`.

Related issue:

- #19164

[Microsoft style guide, describing interactions with UI](https://learn.microsoft.com/en-us/style-guide/procedures-instructions/describing-interactions-with-ui)

**Comment by aslafy-z** [comment]
@HonkingGoose I applied the changes

File: lib/workers/repository/onboarding/pr/__snapshots__/config-description.spec.ts.snap
Comment (HonkingGoose) [inline lib/workers/repository/onboarding/pr/__snapshots__/config-description.spec.ts.snap:12]: How about something like this?

- Use present tense
- Write short sentences
- Use simple words
- Drop words like `simply` (if we need to explain, it's not that simple 😄)

> Do you want to change how Renovate upgrades your dependencies? Put your custom config in the \`.github/renovate.json\` file in this branch and select the Retry/Rebase checkbox below. Renovate will update the list of Pull Requests in the "What to Expect" section the next time it runs.

Comment (aslafy-z) [reply lib/workers/repository/onboarding/pr/__snapshots__/config-description.spec.ts.snap:12]: Additionally, the list of pull requests is not the only thing that can changes. For eg, if you ignore a file, it will be removed from the detected list. We can maybe simplify it like

```
🔡 Do you want to change how Renovate upgrades your dependencies? Put your custom config in the `.github/renovate.json` file in this branch and select the Retry/Rebase checkbox below. Renovate will update the Pull Request description the next time it runs.
```

Comment (aslafy-z) [reply lib/workers/repository/onboarding/pr/__snapshots__/config-description.spec.ts.snap:12]:
```suggestion
🔡 Do you want to change how Renovate upgrades your dependencies? Put your custom config in the `.github/renovate.json` file in this branch and select the Retry/Rebase checkbox below. Renovate will update the Pull Request description the next time it runs.
```

Comment (HonkingGoose) [reply lib/workers/repository/onboarding/pr/__snapshots__/config-description.spec.ts.snap:12]: I like your proposal. I'll let a a maintainer review and apply the suggestion. 😉

Comment (aslafy-z) [reply lib/workers/repository/onboarding/pr/__snapshots__/config-description.spec.ts.snap:12]: cc @rarkins @RahulGautamSingh

Comment (RahulGautamSingh) [reply lib/workers/repository/onboarding/pr/__snapshots__/config-description.spec.ts.snap:12]: I don't think we need to select the checkbox for the PR body to be updated. Can you create a minimal reproduction of this?

**Comment by RahulGautamSingh** [comment]
> I thought that rebase/retry will erase any custom config. @RahulGautamSingh

It used to that way.
But we didn't need to rebase anymore

Context:
In [this PR](https://github.com/renovatebot/renovate/pull/20893) the rebase logic was removed. 
We started merging the onboarding-branch into base-branch. Which meant that we now process the modified PRs as well. 
Hence if the user wants to rebase and update the pr body they just need to add a new commit which revert their changes.

The bug user is facing seems to be something else. I will try to reproduce it and get back asap.

**Comment by aslafy-z** [comment]
@RahulGautamSingh I did some more tests. It looks like renovate behaves like this when `onboardingRebaseCheckbox` is set to `true`. I will remove this value from our configuration, it's still an experimental flag. However, I feel like it would be nice if the message was contextualized depending on the renovate configuration. WDYT?

**Comment by RahulGautamSingh** [comment]
Yes, the message should be contextualized as you say. We also need to change the name `onboardingRebaseCheckbox` to something more accurate like `onboardingPrRefreshCheckbox` since the expected outcome is that the prBody is refreshed and not that the branch is rebased.

@rarkins Thoughts

File: lib/workers/repository/onboarding/pr/config-description.ts
Comment (RahulGautamSingh) [inline lib/workers/repository/onboarding/pr/config-description.ts:55]: I prefet the existing wording.

File: lib/workers/repository/onboarding/pr/config-description.ts
Comment (RahulGautamSingh) [inline lib/workers/repository/onboarding/pr/config-description.ts:61]:
```suggestion
desc += `Simply edit the \`${configFile}\` in this branch with your custom config${config.onboardingRebaseCheckbox ? ' and select the Retry/Rebase checkbox below' : ''}. Renovate will update the Pull Request description the next time it runs.`;
```

File: lib/workers/repository/onboarding/pr/config-description.ts
Comment (aslafy-z) [reply lib/workers/repository/onboarding/pr/config-description.ts:55]: I applied thisesuggestion https://github.com/renovatebot/renovate/pull/25899#discussion_r1401801560.
I can for sure revert it if you want me to

Comment (RahulGautamSingh) [reply lib/workers/repository/onboarding/pr/config-description.ts:55]: It's okay if Honking Goose suggested it. He is better at this 👍

File: lib/workers/repository/onboarding/pr/config-description.spec.ts
Comment (RahulGautamSingh) [inline lib/workers/repository/onboarding/pr/config-description.spec.ts:93]: Not necessary since snapshot check, does the same.

**Review by RahulGautamSingh — APPROVED** [review]

**Comment by RahulGautamSingh** [comment]
Also can you modify the PR title to something like:

`fix(onboarding/pr): only add retry/refresh checkbox if onboardingRebaseCheckbox is true`

**Comment by HonkingGoose** [comment]
Please avoid force pushing. We squash merge the pr later anyways.

File: lib/workers/repository/onboarding/pr/__snapshots__/config-description.spec.ts.snap
Comment (aslafy-z) [inline lib/workers/repository/onboarding/pr/__snapshots__/config-description.spec.ts.snap:66]:
```suggestion
exports[`workers/repository/onboarding/pr/config-description getConfigDesc() include retry/refresh checkbox only if onboardingRebaseCheckbox is true 1`] = `
```

File: lib/workers/repository/onboarding/pr/config-description.spec.ts
Comment (aslafy-z) [inline lib/workers/repository/onboarding/pr/config-description.spec.ts:93]:
```suggestion
```

File: lib/workers/repository/onboarding/pr/config-description.spec.ts
Comment (aslafy-z) [inline lib/workers/repository/onboarding/pr/config-description.spec.ts:86]:
```suggestion
    it('include retry/refresh checkbox message only if onboardingRebaseCheckbox is true', () => {
```

File: lib/workers/repository/onboarding/pr/config-description.spec.ts
Comment (aslafy-z) [reply lib/workers/repository/onboarding/pr/config-description.spec.ts]: Removed the line

**Review by viceice — APPROVED** [review]

**Pull request merged**
